plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusIntegrationMetrics.java

```diff
@@ -13,139 +13,62 @@
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
 import org.testng.annotations.Test;
 
-import static io.trino.plugin.prometheus.PrometheusClient.TIMESTAMP_COLUMN_TYPE;
-import static io.trino.plugin.prometheus.PrometheusQueryRunner.createPrometheusClient;
-import static io.trino.spi.type.BigintType.BIGINT;
-import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
-import static io.trino.testing.TestingConnectorSession.SESSION;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNull;
-import static org.testng.Assert.assertTrue;
+import static io.trino.plugin.prometheus.PrometheusQueryRunner.createPrometheusQueryRunner;
 
-/**
- * Integration tests against Prometheus container
- */
-@Test(singleThreaded = true)
 public class TestPrometheusIntegrationMetrics
+        extends AbstractTestQueryFramework
 {
-    private static final PrometheusTableHandle RUNTIME_DETERMINED_TABLE_HANDLE = new PrometheusTableHandle("default", "up");
-
-    private PrometheusServer server;
-    private PrometheusClient client;
-
-    @BeforeClass
-    public void createQueryRunner()
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
     {
-        this.server = new PrometheusServer();
-        this.client = createPrometheusClient(server);
-    }
-
-    @AfterClass(alwaysRun = true)
-    public final void destroy()
-    {
-        server.close();
+        PrometheusServer server = closeAfterClass(new PrometheusServer());
+        return createPrometheusQueryRunner(server, ImmutableMap.of(), ImmutableMap.of());
     }
 
     @Test
-    public void testRetrieveUpValue()
+    public void testShowTables()
     {
-        assertTrue(client.getTableNames("default").contains("up"), "Prometheus' own `up` metric should be available in default");
+        assertQuery("SHOW TABLES IN default LIKE 'up'", "VALUES 'up'");
     }
 
     @Test
-    public void testHandleErrorResponse()
+    public void testShowCreateSchema()
     {
-        assertThatThrownBy(() -> client.getTableNames("unknown"))
-                .isInstanceOf(TrinoException.class)
-                .hasMessageContaining("Prometheus did no return metrics list (table names)");
-        PrometheusTable table = client.getTable("unknown", "up");
-        assertNull(table);
+        assertQuery("SHOW CREATE SCHEMA default", "VALUES 'CREATE SCHEMA prometheus.default'");
+        assertQueryFails("SHOW CREATE SCHEMA unknown", ".*Schema 'prometheus.unknown' does not exist");
     }
 
     @Test
     public void testListSchemaNames()
     {
-        PrometheusMetadata metadata = new PrometheusMetadata(client);
-        assertEquals(metadata.listSchemaNames(SESSION), ImmutableSet.of("default"));
+        assertQuery("SHOW SCHEMAS LIKE 'default'", "VALUES 'default'");
     }
 
     @Test
-    public void testGetColumnMetadata()
```

Comment:
What was this trying to test? It seems it's just checking if we pass in a handle for an object that doesn't actually exist we still get back metadata for it.

Reply:
Some connectors (e.g. memory) gets tables in `getColumnMetadata()`, so I guess the test tried to show such validation doesn't happen in the connector.